app/src/androidTest/java/org/mozilla/fenix/syncintegration/SyncIntegrationTest.kt

```diff
@@ -0,0 +1,173 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.syncintegration
+
+import android.os.SystemClock.sleep
+import android.widget.Button
+import android.widget.EditText
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.replaceText
+import androidx.test.espresso.action.ViewActions.pressImeActionButton
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.ui.robots.homeScreen
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Until
+import org.hamcrest.Matchers.allOf
+import org.mozilla.fenix.R
+
+@Suppress("RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
+class SyncIntegrationTest {
+    val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    // History item Desktop -> Fenix
+    @Test
+    fun checkHistoryFromDesktopTest() {
+        signInFxSync()
+        tapReturnToPreviousApp()
+        homeScreen {
+        }.openThreeDotMenu {}
+        libraryButton()
+        historyButton()
+        historyAfterSyncIsShown()
+    }
+    /* These tests will be running in the future
+    // once the test above runs successfully and
+    // the environment is stable
+
+    // Bookmark item Desktop -> Fenix
+    @Test
+    fun checkBookmarkFromDesktopTest() {
+        signInFxSync()
+        tapReturnToPreviousApp()
+        sleep(5000)
+        homeScreen {
+        }.openThreeDotMenu {}
+        libraryButton()
+        bookmarkButton()
+        bookmarkAfterSyncIsShown()
+    }
+
+    // History item Fenix -> Desktop
+    @Test
+    fun checkBookmarkFromDeviceTest() {
+        tapInToolBar()
+        typeInToolBar()
+        seeBookmark()
+        mDevice.pressBack()
+        signInFxSync()
+    }
+
+    // Bookmark item Fenix -> Desktop
+    @Test
+    fun checkHistoryFromDeviceTest() {
+        tapInToolBar()
+        typeInToolBar()
+        sleep(3000)
+        mDevice.pressBack()
+        signInFxSync()
+    }
+    */
+
+    // Useful functions for the tests
+    fun typeEmail() {
+        val emailInput = mDevice.findObject(UiSelector()
+                .instance(0)
+                .className(EditText::class.java))
+        emailInput.waitForExists(1000)
```

Comment:
see above

Reply:
I will change that to use the constant in helpers. Sorry I forgot about that.